File: js/baekjoon/step3/bkjn_25304_1.js
const input = require("fs").readFileSync("dev/stdin").toString().split("\n");

let x = Number(input[0]);
// === let x = +input[0]
let n = Number(input[1]);
let total = 0;

for (i = 2; i <= n + 1; i++) {
  let price = input[i].split(" ").map((item) => Number(item));
  total += price[0] * price[1];
}
// for (i = 2; i <= n + 1; i++) {
//   let price = input[i].split(" ");
//   total += Number(price[0]) * Number(price[1]);
// }

console.log(x === total ? "Yes" : "No");

File: python/baekjoon/step3/bkjn_25304_1.py
total_price = int(input())
total_type = int(input())

add_price = 0

for i in range(total_type):
  price, number = map(int, input().split())
  add_price += price * number

if total_price == add_price:
  print("Yes")
else:
  print("No")